frontend/src/v5/ui/routes/dashboard/projects/federations/federationSettingsForm/federationSettingsForm.component.tsx

```diff
@@ -0,0 +1,405 @@
+	const {
+		latLong,
+		position,
+	} = federation.settings?.surveyPoint || {};
+	const [x, y, z] = position || [0, 0, 0];
+	const [latitude, longitude] = latLong || [0, 0];
+	const { code, name, description: desc } = federation;
+	const defaultView = federation?.settings?.defaultView || EMPTY_VIEW._id;
+	return {
+		name,
+		desc,
+		code,
+		unit,
+		defaultView,
+		latitude,
+		longitude,
+		angleFromNorth,
+		x,
+		y,
+		z,
+	};
+};
+
+const FederationSchema = Yup.object().shape({
+	name: Yup.string()
+		.min(2,
+			formatMessage({
+				id: 'federations.settings.name.error.min',
+				defaultMessage: 'Federation Name must be at least 2 characters',
+			}))
+		.max(120,
+			formatMessage({
+				id: 'federations.settings.name.error.max',
+				defaultMessage: 'Federation Name is limited to 120 characters',
+			}))
+		.required(
+			formatMessage({
+				id: 'federations.settings.name.error.required',
+				defaultMessage: 'Federation Name is a required field',
+			}),
+		),
+	desc: Yup.lazy((value) => (
+		value === ''
+			? Yup.string().strip()
+			: Yup.string()
+				.min(1,
+					formatMessage({
+						id: 'federations.settings.description.error.min',
+						defaultMessage: 'Federation Description must be at least 1 character',
+					}))
+				.max(600,
+					formatMessage({
+						id: 'federations.settings.description.error.max',
+						defaultMessage: 'Federation Description is limited to 600 characters',
+					}))
+	)),
+	unit: Yup.string().required().default('mm'),
+	code: Yup.lazy((value) => (
+		value === ''
+			? Yup.string().strip()
+			: Yup.string()
+				.min(1,
+					formatMessage({
+						id: 'federations.settings.code.error.min',
+						defaultMessage: 'Code must be at least 1 character',
+					}))
+				.max(50,
+					formatMessage({
+						id: 'federations.settings.code.error.max',
+						defaultMessage: 'Code is limited to 50 characters',
+					}))
+				.matches(/^[\w|_|-]*$/,
+					formatMessage({
+						id: 'federations.settings.code.error.characters',
+						defaultMessage: 'Code can only consist of letters and numbers',
+					}))
+	)),
+	defaultView: Yup.string()
+		.nullable()
+		.transform((value) => (value === EMPTY_VIEW._id ? null : value)),
+	latitude: Yup.number().required(),
+	longitude: Yup.number().required(),
+	angleFromNorth: Yup.number()
+		.min(0,
+			formatMessage({
+				id: 'federations.settings.angle.error.min',
+				defaultMessage: 'Angle cannot be smaller than 0',
+			}))
+		.max(360,
+			formatMessage({
+				id: 'federations.settings.angle.error.max',
+				defaultMessage: 'Angle cannot be greater than 360',
+			}))
+		.transform((value) => value ?? 0),
+	x: Yup.number().required(),
+	y: Yup.number().required(),
+	z: Yup.number().required(),
+});
+
+const getThumbnailBasicPath = (teamspace: string, projectId: string, federationId: string) => (
+	(viewId: string) => `teamspaces/${teamspace}/projects/${projectId}/federations/${federationId}/views/${viewId}/thumbnail`
+);
```

Comment:
This function is not returning what it should. The url should be constructed like in https://github.com/3drepo/3drepo.io/blob/87c28589191861d282a4a9eff237132411058bcf/frontend/src/v5/services/api/default.ts#L69

also we could refactor a bit of 
`generateV5ApiUrl(url)` to  `generateV5ApiUrl(url, requestMethod)` 

like this:
`const generateV5ApiUrl = (url: string, requestMethod: string): string => encodeURI(clientConfigService.apiUrl(requestMethod, `v5/${url}`));
`
and then be used like this:
```
const getRequest = (url, ...options) => {
	const requestUrl = generateV5ApiUrl(url, clientConfigService.GET_API);
	return axios.get(requestUrl, ...options);
};
```

In getThumbnailBasicPath would be used like this:

```
const getThumbnailBasicPath = (teamspace: string, projectId: string, federationId: string) => (
	(viewId: string) => generateV5ApiUrl(`teamspaces/${teamspace}/projects/${projectId}/federations/${federationId}/views/${viewId}/thumbnail`, clientConfigService.GET_API);
````

Reply:
Done